Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -276,7 +314,7 @@ extension AWSClient {
                 .applyMiddlewares(serviceConfig.middlewares + self.middlewares)
                 .createHTTPRequest(signer: signer)
         }.flatMap { request in
-            return self.invoke(request, with: serviceConfig, on: eventLoop, logger: logger)
+            return self.invoke(request, with: serviceConfig, on: eventLoop, logger: logger, context: context)
```

Comment:
this will be added per "// TODO: create internal spans for signing, parsing?"

there are quite a many public function a just 2 internal invoke, so that was a starting point

Reply:
Ah missed the parsing. I was going to create a central execute function which all the others call. I could do that now if you want. Will create a painful merge but might make things easier

Reply:
#349

Reply:
copied your change "manually", merging different incompatible API is going to be a huge hassle